### PR TITLE
Fix for WFCORE-1726. Support for attachments in Response.

### PR DESCRIPTION
**Pull request by jfdenise**
A new attachment command to download/display attachments. Evolved a batch command with an optional ResponseHandler that is called for each step once the composite reply has been received.

### CODE REVIEW

**Comment by bstansberry** [comment]
retest this please

**Comment by bstansberry** [comment]
@aloubyansky please approve

**Comment by aloubyansky** [comment]
Looks good. I'd think that in non-batch mode the target file should be overridden. But perhaps there are good reasons to use the index instead.

**Comment by jfdenise** [comment]
@aloubyansky, in non batch mode, in order to preserve all the received streams, I implemented the file indexing. With file overriding we could confuse users that would expect multiple saved file.

**Comment by aloubyansky** [comment]
That makes sense. What I meant though was when you execute the same operation consequently, it does not override the file specified in the argument.
This is not an objection. That would be just my first expectation.

**Comment by jfdenise** [comment]
I see, calling multiple time the same operation. 
I was on the safe side, not alter any user data. If a user file with the same name already exists, I would not override it. Ideally, in a UI context I would ask the user for override or copy.

We could had the option --override to the command (used in conjunction with save) in order to override the file explicitly.

**Comment by aloubyansky** [comment]
It all goes down to user expectation. One thing is to safe as you meant it. Another thing is to have the output at the location specified. I think it could be more important in case of automation/scripts when the output is then consumed and processed.

**Comment by jfdenise** [comment]
I agree, scripts would expect deterministic paths for files. Your experience in this matter makes me feel that we need the override option. Do you think that it would answer the scenarios you are seeing? If yes, then I would introduce the option. We would be safe by default, and override would be explicit. 
Perhaps that you would expect the opposite ;-)?

**Comment by aloubyansky** [comment]
The override option would surely cover the issue I was thinking about. I don't have anything against the safe default. That's actually a great feature and pretty clean implementation. Thanks, Jeff!

**Comment by jfdenise** [comment]
Thanks @aloubyansky for your great feedbacks.

**Comment by bstansberry** [comment]
@aloubyansky please @ mention me when this looks good to you. I'll put a fixme label on it now to stop me checking it. The EAP7 is through analysis. :)

File: cli/src/main/java/org/jboss/as/cli/handlers/AttachmentHandler.java [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+                            name = name.substring(0, indexed);
+                        } catch (NumberFormatException ex) {
+                            // XXX OK, not a number.
+                        }
+                    } else {
+                        index += 1;
+                    }
+                    targetFile = new File(name + "(" + index + ")");
+                }
+            }
```
Comment (aloubyansky) [inline cli/src/main/java/org/jboss/as/cli/handlers/AttachmentHandler.java]: If override is true and the response contains multiple streams, shouldn't the index be incremented here?

Comment (jfdenise) [inline cli/src/main/java/org/jboss/as/cli/handlers/AttachmentHandler.java]: Yes, well spotted. PR updated.

**Comment by aloubyansky** [comment]
@bstansberry this is good to merge, thanks!

**Comment by jfdenise** [comment]
Documenting the command, I realise that override should be replaced by overwrite, override doesn't apply well to file. I am making the change and re-sync the PR.

**Comment by bstansberry** [comment]
retest this please

**Pull request merged**